examples/nestjs/consumer/test/app.spec.ts

```diff
@@ -0,0 +1,267 @@
+
+describe('Pact', () => {
+  let pactFactory: PactFactory;
+  let provider: Pact;
+  let animalsService: AppService;
+
+  beforeAll(async () => {
+    const moduleRef = await Test.createTestingModule({
+      imports: [AppModule, PactModule],
+    }).compile();
+
+    pactFactory = moduleRef.get(PactFactory);
+
+    provider = pactFactory.createContractBetween({
+      consumer: 'Matching Service',
+      provider: 'Animal Profile Service',
+    });
+
+    animalsService = moduleRef.get(AppService);
+
+    /*
+     * Setup a Mock Server before unit tests run.
+     * This server acts as a Test Double for the real Provider API.
+     * We then call addInteraction() for each test to configure the Mock Service
+     * to act like the Provider
+     * It also sets up expectations for what requests are to come, and will fail
+     * if the calls are not seen.
+     */
+    const { port } = await provider.setup();
+    process.env.API_HOST = `http://localhost:${port}`;
+  });
+
+  // Alias flexible matchers for simplicity
+  const { eachLike, like, term, iso8601DateTimeWithMillis } = Matchers;
+
+  // Animal we want to match :)
+  const suitor: Animal = {
+    id: 2,
+    available_from: new Date('2017-12-04T14:47:18.582Z'),
+    first_name: 'Nanny',
+    animal: 'goat',
+    last_name: 'Doe',
+    age: 27,
+    gender: 'F',
+    location: {
+      description: 'Werribee Zoo',
+      country: 'Australia',
+      post_code: 3000,
+    },
+    eligibility: {
+      available: true,
+      previously_married: true,
+    },
+    interests: ['walks in the garden/meadow', 'parkour'],
+  };
+
+  const MIN_ANIMALS = 2;
+
+  /*
+   * Define animal payload, with flexible matchers
+   * This makes the test much more resilient to changes in actual data.
+   * Here we specify the 'shape' of the object that we care about.
+   * It is also import here to not put in expectations for parts of the
+   * API we don't care about
+   */
+  const animalBodyExpectation = {
+    id: like(1),
+    available_from: iso8601DateTimeWithMillis(),
+    first_name: like('Billy'),
+    last_name: like('Goat'),
+    animal: like('goat'),
+    age: like(21),
+    gender: term({
+      matcher: 'F|M',
+      generate: 'M',
+    }),
+    location: {
+      description: like('Melbourne Zoo'),
+      country: like('Australia'),
+      post_code: like(3000),
+    },
+    eligibility: {
+      available: like(true),
+      previously_married: like(false),
+    },
+    interests: eachLike('walks in the garden/meadow'),
+  };
+
+  // Define animal list payload, reusing existing object matcher
+  const animalListExpectation = eachLike(animalBodyExpectation, {
+    min: MIN_ANIMALS,
+  });
+
+  // After each individual test (one or more interactions)
+  // we validate that the correct request came through.
+  // This ensures what we _expect_ from the provider, is actually
+  // what we've asked for (and is what gets captured in the contract)
+  afterEach(() => provider.verify());
```

Comment:
This is specified twice, I think - it's also at the bottom. You only need once (and with `jest-pact`, you don't need it at all).

Reply:
Resolved in the other branch